test/integration/kubernetes_deploy_test.rb

```diff
@@ -1,27 +1,134 @@
+
+    deploy_fixture_set("basic")
+    basic.assert_all_up
+  end
+
+  def test_pruning
+    deploy_fixture_set("basic")
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_up
+
+    deploy_fixture_set("basic", subset: ["redis"])
+    basic.assert_all_redis_resources_up
+    basic.refute_configmap_data_exists
+    basic.refute_managed_pod_exists
+    basic.refute_web_resources_exist
+  end
+
+  def test_pvcs_are_not_pruned
+    deploy_fixture_set("basic", subset: ["redis"])
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_redis_resources_up
+
+    deploy_fixture_set("basic", subset: ["configmap-data"])
+    basic.assert_configmap_data_present
+    basic.refute_redis_resources_exist(expect_pvc: true)
+  end
+
+  def test_success_with_unrecognized_resource_type
+    # Secrets are intentionally unsupported because they should not be committed to your repo
+    fixture_set = load_fixture_data("basic", ["configmap-data"])
+    secret = {
+      "apiVersion" => "v1",
+      "kind" => "Secret",
+      "metadata" => { "name" => "test" },
+      "data" => { "foo" => "YmFy" }
+    }
+    fixture_set["secret"] = { "Secret" => secret }
+    deploy_loaded_fixture_set(fixture_set)
+
+    live_secret = kubeclient.get_secret("test", @namespace)
+    assert_equal({ foo: "YmFy" }, live_secret["data"].to_h)
+  end
+
+  def test_invalid_yaml_fails_fast
+    error = assert_raises(KubernetesDeploy::FatalDeploymentError) do
+      deploy_fixture_set("invalid", subset: ["yaml-error"])
+    end
+    assert_match /Template \S+ cannot be parsed/, error.to_s
+  end
+
+  def test_invalid_k8s_spec_that_is_valid_yaml_fails_fast
+    fixture_set = load_fixture_data("basic", ["configmap-data"])
+    configmap = fixture_set["configmap-data"]["ConfigMap"].first
+    configmap["metadata"]["myKey"] = "uhOh"
+
+    error = assert_raises(KubernetesDeploy::FatalDeploymentError) do
+      deploy_loaded_fixture_set(fixture_set)
+    end
+    assert_match /Dry run failed for template configmap-data/, error.to_s
+
+    @logger_stream.rewind
+    assert_match /error validating data\: found invalid field myKey for v1.ObjectMeta/, @logger_stream.read
+  end
+
+  def test_dead_pods_in_old_replicaset_are_ignored
+    fixture_set = load_fixture_data("basic", ["configmap-data", "web"])
+    deployment = fixture_set["web"]["Deployment"].first
+    container = deployment["spec"]["template"]["spec"]["activeDeadlineSeconds"] = 1
+    deploy_loaded_fixture_set(fixture_set, wait: false) # this will never succeed as pods are killed after 1s
+
+    sleep 1 # make sure to hit DeadlineExceeded on at least one pod
+
+    deploy_fixture_set("basic", subset: ["web", "configmap-data"])
+    pods = kubeclient.get_pods(namespace: @namespace, label_selector: "name=web,app=basic")
+    running_pods, not_running_pods = pods.partition { |pod| pod.status.phase == "Running" }
+    assert_equal 1, running_pods.size
+    assert not_running_pods.size >= 1
```

Comment:
I don't think so, unless I'm misunderstanding for what... `basic.assert_some_web_pods_running(running: N, min_not_running: M)` doesn't strike me as clearer or reusable.